src/renderer/components/hotbar/hotbar-switch-command.tsx

```diff
@@ -25,62 +25,81 @@ interface Dependencies {
   commandOverlay: CommandOverlay;
 }
 
-function ignoreIf<T>(check: boolean, menuItems: T) {
+function ignoreIf<T>(check: boolean, menuItems: T[]): T[] {
   return check ? [] : menuItems;
 }
 
-function getHotbarSwitchOptions(hotbars: Hotbar[]): (Hotbar | typeof hotbarAddAction | typeof hotbarRemoveAction | typeof hotbarRenameAction)[] {
+interface HotbarSwitchActionOption {
+  action: typeof hotbarAddAction | typeof hotbarRemoveAction | typeof hotbarRenameAction;
+}
+
+interface SwitchToHotbarOption {
+  hotbar: Hotbar;
+}
+
+type HotbarSwitchOption = SwitchToHotbarOption | HotbarSwitchActionOption;
+
+function getHotbarSwitchOptions(hotbars: Hotbar[]): HotbarSwitchOption[] {
   return [
-    ...hotbars,
-    hotbarAddAction,
+    ...hotbars.map(hotbar => ({ hotbar })),
+    { action: hotbarAddAction },
     ...ignoreIf(hotbars.length > 1, [
-      hotbarRemoveAction,
-    ] as const),
-    hotbarRenameAction,
+      { action: hotbarRemoveAction } as const,
+    ]),
+    { action: hotbarRenameAction },
   ];
 }
 
+function isActionOption(option: HotbarSwitchOption): option is HotbarSwitchActionOption {
+  return Boolean((option as HotbarSwitchActionOption).action);
+}
+
 const NonInjectedHotbarSwitchCommand = observer(({
   hotbarStore,
   commandOverlay,
 }: Dependencies) => (
   <Select
     id="switch-to-hotbar-input"
     menuPortalTarget={null}
-    onChange={(value) => {
-      switch (value) {
-        case hotbarAddAction:
-          return commandOverlay.open(<HotbarAddCommand />);
-        case hotbarRemoveAction:
-          return commandOverlay.open(<HotbarRemoveCommand />);
-        case hotbarRenameAction:
-          return commandOverlay.open(<HotbarRenameCommand />);
+    onChange={(option) => {
+      if (!option) {
+        return;
+      }
 
-        default: {
-          if (value) {
-            hotbarStore.setActiveHotbar(value);
-            commandOverlay.close();
-          }
+      if (isActionOption(option)) {
+        switch (option.action) {
+          case hotbarAddAction:
+            return commandOverlay.open(<HotbarAddCommand />);
+          case hotbarRemoveAction:
+            return commandOverlay.open(<HotbarRemoveCommand />);
+          case hotbarRenameAction:
+            return commandOverlay.open(<HotbarRenameCommand />);
```

Comment:
Big nit: Open Closed Principle violation. Instead of `switch`, there could ideally be something like: `option.action(commandOverlay)`, or better yet, `option.action()`, where the action can inject `commandOverlay` and/or more where needed.

Reply:
This is a single command (and business logic), not the entire command overlay. I don't see why OCP even matters here.

Reply:
Most of all it matters as a policy I'd warmly recommend you to welcome :)

It also matters, as very often code like this starts humble, and then grows to the wrong tracks by not having visibility for things such as OCP early on.

Reply:
Then if necessary once it gets large enough, we can change it.